vault/vault.go

```diff
@@ -218,7 +218,7 @@ func (t *tempCredsCreator) provider(config *Config) (aws.CredentialsProvider, er
 		if err != nil {
 			return nil, err
 		}
-	} else if config.HasSSOStartURL() {
+	} else if config.HasSSOStartURL() || config.HasSSOSession() {
 		return NewSSORoleCredentialsProvider(t.keyring.Keyring, config)
```

Comment:
In proper case, both of them are true. However, `[sso-session]` may lack `sso_start_url`. In that case I feel like it should still go the SSO way.